Problems building: undefined_symbol: sodium_init

I was looking for emacs keepassxc integration and came across this and keepassxc.el, but am having trouble building.

First, I let `straight` fetch this for me via:
```
(use-package sodium
  :straight (:host github :repo "dakra/sodium.el"
                   :branch "master"))
```
which failed with:
```
Debugger entered--Lisp error: (file-missing "Cannot open load file" "No such file or directory" "libsodium")
  require(libsodium)
```

So then in a terminal I hopped into the repo directory and:
```
$ make
emacs -Q -batch -L . -f batch-byte-compile sodium.el
emacs: symbol lookup error: /home/len/.emacs.d/straight/repos/sodium.el/libsodium.so: undefined symbol: sodium_init
Makefile:19: recipe for target 'sodium.elc' failed
make: *** [sodium.elc] Error 127
```

Other info:
```
$  lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.3 LTS
Release:        18.04
Codename:       bionic
$ emacs --version
GNU Emacs 26.3
[...]
$ dpkg -l libsodium-dev
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                                        Version                    Architecture               Description
+++-===========================================-==========================-==========================-============================================================================================
ii  libsodium-dev:amd64                         1.0.16-2                   amd64                      Network communication, cryptography and signaturing library - headers
```

Any ideas?

Sorry, that make output I posted was from a re-attempt. If I make clean and rerun I do get the cc command, but the emacs command is still failing:
```
$ make
cc -shared -std=c99 -s -Wall -Wextra -O3 -fpic -lsodium -o libsodium.so libsodium.c
emacs -Q -batch -L . -f batch-byte-compile sodium.el
emacs: symbol lookup error: /home/len/.emacs.d/straight/repos/sodium.el/libsodium.so: undefined symbol: sodium_init
Makefile:19: recipe for target 'sodium.elc' failed
make: *** [sodium.elc] Error 127
```
There is a libsodium there:
```
$ ll
total 92
-rw-rw-r-- 1 len len 11403 Oct 26 13:15 emacs-module.h
-rw-rw-r-- 1 len len  9459 Oct 26 13:15 libsodium.c
-rwxrwxr-x 1 len len 14416 Oct 26 13:16 libsodium.so*
-rw-rw-r-- 1 len len 35147 Oct 26 13:15 LICENSE
-rw-rw-r-- 1 len len   825 Oct 26 13:15 Makefile
-rw-rw-r-- 1 len len  1744 Oct 26 13:15 README.md
-rw-rw-r-- 1 len len  1082 Oct 26 13:15 sodium-box-demo.el
-rw-rw-r-- 1 len len  1276 Oct 26 13:15 sodium.el
```
It just looks like the emacs command isn't able to connect from the newly built libsodium.so there to the actual system sodium library in /usr/lib/x86_64-linux-gnu/libsodium.so  (is it getting confused by the fact they have the same name?)

In terms of my use case, I have several secrets (e.g. passwords, GitHub API tokens for grip-mode, etc) currently sitting in my init.el that I want to move out. It sounds like emacs auth-source deals with that, but I was hoping there would be an auth-source backend that uses keepassxc (since that is what I use for my other password storage) rather than having to make a new location for storing passwords.

--- a/Makefile
+++ b/Makefile
@@ -10,7 +10,7 @@ linux: libsodium.so sodium.elc
 windows: libsodium.dll sodium.elc
 
 libsodium.so: libsodium.c
-	$(CC) -shared $(CFLAGS) $(LDFLAGS) -o $@ $^
+	$(CC) -shared $(CFLAGS) $^ $(LDFLAGS) -o $@
 
 libsodium.dll: libsodium.c
 	$(MINGW_CC) -shared $(CFLAGS) $(LCDFLAGS) -o $@ $^

--- a/libsodium.c
+++ b/libsodium.c
@@ -117,6 +117,7 @@ box_keypair(emacs_env *env, ptrdiff_t n, emacs_value *args, void *ptr)
   fun_args[0] = pk_pair;
   fun_args[1] = sk_pair;
 
+  list = env->intern(env, "list");
   return env->funcall(env, list, 2, fun_args);
 }
 

